agent.go

```diff
@@ -1135,19 +1135,18 @@ func (a *Agent) handleInbound(m *stun.Message, local Candidate, remote net.Addr)
 
 // validateNonSTUNTraffic processes non STUN traffic from a remote candidate,
 // and returns true if it is an actual remote candidate
-func (a *Agent) validateNonSTUNTraffic(local Candidate, remote net.Addr) bool {
-	var isValidCandidate uint64
+func (a *Agent) validateNonSTUNTraffic(local Candidate, remote net.Addr) (Candidate, bool) {
+	var remoteCandidate Candidate
 	if err := a.run(local.context(), func(ctx context.Context, agent *Agent) {
```

Comment:
I am wondering if we really need to run this in the agents task loop?
It seem to be a hot path and we might see even better performance improvements if we protect the cache with an `sync.RWMutex`

Reply:
A nevermind, I see that you already do the cache lookup outside the task loop.
But i am a bit concerned if this is really thread safe?